scripts/diagnostic.js

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2023-2024 SolarWinds Worldwide, LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+const path = require("path")
+const fs = require("fs")
+
+function packageJson(id) {
+  try {
+    const directory = require.resolve(id)
+    const file = path.join(directory, "package.json")
+    const contents = fs.readFileSync(file, { encoding: "utf-8" })
+    return JSON.parse(contents)
+  } catch {
+    return null
+  }
+}
+
+function print(...vals) {
+  for (let i = 0; i < vals.length; i++) {
+    console.dir(vals[i], {
+      depth: Infinity,
+      maxArrayLength: Infinity,
+      maxStringLength: Infinity,
+    })
+  }
+}
+
+const report = process.report.getReport()
+const packages = [
+  "solarwinds-apm",
+  "@solarwinds-apm/sdk",
+  "@solarwinds-apm/bindings",
+  "solarwinds-apm-bindings",
+  "@opentelemetry/api",
+  "@opentelemetry/core",
+  "appoptics-apm",
+  "@appoptics/apm-bindings",
+  "appoptics-bindings",
+].reduce((ps, p) => ({ ...ps, [p]: packageJson(p) }), {})
+
+print(report, packages)
+
+const installed = packages["solarwinds-apm"] != null
+if (!installed) {
+  console.warn(
+    "The 'solarwinds-apm' package could not be found.",
+    "Is it installed and are you running this script from your application directory ?",
+  )
+}
+
+const appopticsInstalled = packages["appoptics-apm"] != null
+if (appopticsInstalled) {
+  console.warn(
+    "The 'appoptics-apm' package was detected.",
+    "Make sure to uninstall it properly as it is not compatible with the 'solarwinds-apm' package.",
```

Comment:
Could maybe console log a link to the [Upgrade docs](https://documentation.solarwinds.com/en/success_center/observability/content/configure/services/nodejs/upgrade.htm) for more details, but it might not look as nice. Take or leave 😃